registry/storage/registry.go

```diff
@@ -13,24 +14,27 @@ type registry struct {
 	driver    storagedriver.StorageDriver
 	pm        *pathMapper
 	blobStore *blobStore
+	redis     *redis.Pool
 }
 
 // NewRegistryWithDriver creates a new registry instance from the provided
 // driver. The resulting registry may be shared by multiple goroutines but is
 // cheap to allocate.
-func NewRegistryWithDriver(driver storagedriver.StorageDriver) distribution.Registry {
-	bs := &blobStore{}
+func NewRegistryWithDriver(driver storagedriver.StorageDriver, redis *redis.Pool) distribution.Registry {
```

Comment:
Using `*redis.Pool` seems rather exposed.  Perhaps a factory interface to get a layer cache would be appropriate here.  Though the problem I see is the use of `*blobStore` in the layer info cache.